### DISCUSSION TITLE
Errors in aging model + discrepancy in OKane2022 beta parameter

### DISCUSSION THREAD

**Discussion by MirAbbasAli2A** [pybamm-team/PyBaMM#3505] [Q&A] [answered]
Hello PyBamm Team - 

I am running an aging model considering all aging mechanisms. I have extended the solvent diffusivity within the SEI to be temperature dependent with the arrhenius relationship and multiplied the LAM proportionality constant (beta) by 20 at the anode and by 1 at the cathode. When I perform this setup, I get an IDAconv fail error. Below is a working example of my code:

```python3
#Import necessary libraries
import pybamm
import matplotlib.pyplot as plt
import numpy as np
from os import path
import pandas as pd
from tec_reduced_model.set_parameters import (
    set_thermal_parameters,
    set_experiment_parameters,
    set_ambient_temperature,
    )
from tec_reduced_model.process_experimental_data import import_thermal_data, get_idxs
from pybamm import Parameter, constants, exp

model = pybamm.lithium_ion.DFN(options = {
    "thermal":"lumped",
    "dimensionality":0,
    "cell geometry": "arbitrary", 
    "SEI": "solvent-diffusion limited",
    "SEI porosity change": "true",
    "particle mechanics": ("swelling and cracking", "swelling and cracking"),
    "SEI on cracks": "true",
    "loss of active material": "stress-driven",
    "calculate discharge energy": "true",
    "lithium plating": "partially reversible",
    "lithium plating porosity change": "true"
    },
    name="TDFN",
    )

param = pybamm.ParameterValues("OKane2022")

param = set_thermal_parameters(param, 16, 2.32e6, temperature)

param["Ambient temperature [K]"] = 298.15
param["Initial temperature [K]"] = 298.15
param["Negative electrode LAM constant proportional term [s-1]"] = 20*2.7778e-07
param["Positive electrode LAM constant proportional term [s-1]"] = 2.7778e-07

T = model.variables["Volume-averaged cell temperature [K]"]
 
def solvent_diffusivity_sei(T):
    Ea = 37000
    arrhenius = exp(Ea / constants.R * (1 / 298.15 - 1 / T))
    return 2.5000000000000002e-22 * arrhenius
 
param.update({"Outer SEI solvent diffusivity [m2.s-1]": solvent_diffusivity_sei(T)})

pybamm.set_logging_level("NOTICE")

#Simple cycling experiment
N=550
 
cccv_experiment = pybamm.Experiment([
    ("Charge at 1C until 4.2V",
    "Hold at 4.2V until C/20",
    "Discharge at 1C until 2.5V",
    "Hold at 2.5V until C/20",
    )
    ]*N
    )

#Refine mesh
var_pts = {
    "x_n" : 20,  # x-direction, length, negative electrode
    "x_s" : 20,  # x-direction, length, separator
    "x_p" : 20,  # x-direction, length, positive electrode
    "r_n" : 30,  # number of volumes in the radial direction, negative particle
    "r_p" : 30   # number of volumes in the radial direction, positive particle
}

#Run model
sim = pybamm.Simulation(model, experiment=cccv_experiment,parameter_values=param, var_pts=var_pts)
sol = sim.solve(save_at_cycles= 10)
```

It is worth noting that without multiplying beta by 20, the simulation runs just fine. I also wanted to know what the actual value of beta is? In the OKane2022 paper it is 0.001 but in the paramater set it is 2.7778e-07. When I run with 0.001, the code fails after 20 cycles with a Linear solver did not converge error. 

FYI, here is the error code I am getting:

![image](https://github.com/pybamm-team/PyBaMM/assets/149210587/3e10ecf8-a6c6-46be-b1f6-02a26c5f0d02)

As you can see, it fails at 307 cycles. 

@DrSOKane , @tinosulzer, @brosaplanella I would be truly grateful if you can suggest anything to help fix this problem I am having with my model.

Thanks in advance.

**Comment by MirAbbasAli2A** [comment]
Update: I refined the mesh to have 30 pts for x_n, x_s and x_p as well and it ran till 449 cycles. However, it then failed with the same exact error as before. Any advice/suggestions on how I can fix this?

**Reply by MirAbbasAli2A** [reply]
Another update: it runs successfully if I remove the cracking option for the cathode per the coupled degradation notebook. Are there any issues with trying to model particle cracking in the cathode? Wouldn't it just be calculating the loss in solid phase volume fraction just like at the anode?

**Comment by DrSOKane** [comment, marked as answer]
Hi @MirAbbasAli2A, I'm pleased to see you managed to resolve the problem on your own. Now I'm here, I can give you a detailed explanation of why that solution worked.

There is no point using the `"swelling and cracking"` option in the positive electrode if you're modelling degradation, because there is no SEI to grow on the crack. The option is available if you're interested in studying how the crack propagates within the positive electrode particles, but it does not affect the degradation behaviour. (It also allows SEI on cracks to be enabled in graphite half-cells, where the graphite is the positive electrode, but you don't specify it as a tuple for that case anyway, because the particles mechanics model isn't applicable to lithium metal electrodes, which has no particles.)

The crack model solves Paris' Law of fatigue crack propagation. The solution to Paris' Law is the crack length grows exponentially with time. Exponential growth poses numerical difficulties, so Paris' Law should only be solved where necessary.

The `"swelling only"` model is much easier to solve, and still gives you the loss of solid phase volume fraction, which is what you're interested in for the positive electrode.

**Reply by DrSOKane** [reply]
I can also answer your question about the LAM proportional constant. The simulations for the 2022 paper were perfomed using a legacy version of PyBaMM. In the legacy version, the LAM constant had no dimensions. In the current version, it has dimensions of `[s-1]`. To convert from the old value to the new one, simple divide by 3600 s. So you've done the right thing by multiplying the new value by 20.

**Reply by MirAbbasAli2A** [reply]
@DrSOKane, thank you so much for your replies this makes a lot of sense now. Basically, swelling and cracking should only be used if film is going to grow on the cracks. I got it. 

While I have you here, I hope you don't mind me asking you one more question:
I have extended your model to apply an arrhenius relationship to the solvent diffusivity through the SEI layer. I am trying to use some of the higher Dsol values (e.g. 7.5e-21 1.25e-20) as my reference diffusivity in the equation but I keep getting a solver convergence fail error. My hypothesis is that this is because increasing the reference diffusivity causes added stiffness to the model which causes issues for the solver. I have changed rtol to 1e-3 to help this issue to no avail. Do you have any thoughts on how I can apply this change in the model? 

FYI, the reason I am trying to increase Dsol is because I am trying to increase the amount of degradation. I found experimental data for the LGM50 cell and apparently, at 1C CC-CV charge and discharge, it degrades quite a bit (See below:)

![image](https://github.com/pybamm-team/PyBaMM/assets/149210587/9d96ef79-6b2f-4866-9577-922abb7ea54c)

This too after maxing out the lithium decay constant, beta term, 50x cracking rate in the anode. I would really appreciate it if you have any thoughts on how to really maximize the degradation while maintaining numerical stability. 

Thank you again for your support.

**Reply by DrSOKane** [reply]
Thank you very much @MirAbbasAli2A for sharing your results. 1C charge is not approved by the manufacturer so it's not surprising that it degrades so quickly. What temperature are you charging at?

**Reply by MirAbbasAli2A** [reply]
@DrSOKane, I am charging at an ambient temperature of 25 deg C (which is what the experiment is run at as well). It is worth noting that I am using a lumped thermal model so the temperature would fluctuate during charging.

**Reply by DrSOKane** [reply]
I'm trying to figure out what mechanism could be responsible for these results. There aren't many data points. 

The degradation is approximately linear until about 400 cycles. [This behaviour has been observed before](https://iopscience.iop.org/article/10.1149/2.0281914jes), but at 5 degrees instead of 25. I've tried and failed to reproduce that result in PyBaMM, which I suspect is because the charging protocol used in that experiment has not been implemented in PyBaMM.

Do you have any other data, such as resistance, degradation mode analysis, or EIS? Or similar results for other temperatures?

**Reply by MirAbbasAli2A** [reply]
@DrSOKane, I appreciate you taking the time to help me understand how these results came to be.
 
I have attached the paper that generated these results for your reference. Figure 3b gives a map of resistance at 50% SOC. 
[s00502-020-00814-9 (4).pdf](https://github.com/pybamm-team/PyBaMM/files/13348215/s00502-020-00814-9.4.pdf)

I was trying to run the first test in Table 2. 

For your reference, here is a poster giving results for the other C-rate conditions:

https://cidetec.es/imodbatt/en/wp-content/uploads/2019/11/A3PS-WS_iModBat.pdf

As far as EIS goes, I recently found this SOH dataset that also has EIS performed for RPT checks. The testing protocol was slightly different however:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10293954/

**Reply by DrSOKane** [reply]
According to the poster, cycling at the same rate but with only 60% DoD means the knee point is avoided and the cell lasts a long time. Which SoC range is avoided? The high one? Or the low one?

**Reply by MirAbbasAli2A** [reply]
@DrSOKane , they actually reduce the discharge window from both ends. So instead of discharging from 4.2V to 2.5V, they discharge from 3.98V to 3.42V. 

See this table from the paper I attached in the previous reply:

![image](https://github.com/pybamm-team/PyBaMM/assets/149210587/f71b4d3c-32bb-4ea5-a37d-59913c9ef05b)

**Reply by MirAbbasAli2A** [reply]
@DrSOKane, I thought I would let you know that I got good correlation to the data set at a 1C discharge rate at 80pDOD and 100pDOD:

![image](https://github.com/pybamm-team/PyBaMM/assets/149210587/e9850902-b321-4e96-be65-1fa3277371c1)

This was done by multiplying the cracking rate by 50, applying an arrhenius correlation to the SEI solvent diffusivity of 2.5E-21, a decay constant of 2.5E-6 and MOST importantly, changing the LAM beta constant of the anode to 80 for the 100pDOD case and 70 for the 80PDOD case. I found that the LAM beta factor makes the most significant change. 

However when I try a beta factor of 50 for the anode (because all other values seem to fail) for a 60pDOD case (keeping everything else constant from the previous runs), then the correlation is not so great:

![image](https://github.com/pybamm-team/PyBaMM/assets/149210587/e2fd0151-ccef-469a-a387-a2c4c123996f)

Are the model parameters meant to be adjusted for every test condition or are they meant to stay constant? If they are meant to be adjusted, in the absence of experimental data, is this just meant to be done intuitively (i.e. low c-rates and depth of discharge means limited plating and LAM? Or is there a general guideline to tune these parameters according to the input conditions?

**Reply by DrSOKane** [reply]
You should not adjust the model parameters for each test condition. The test of a truly accurate parameter set is its ability to predict correct results for the same cell under different conditions.

When you stop to consider that lithium-ion batteries are known to have more degradation mechanisms than the ones currently included in PyBaMM, the quality of the results you have shown is impressive.

Another limitation of the OKane2022 parameter set is that the solid-phase diffusivities are functions of temperature only, not concentration. This will have a large impact on the particle mechanics models. Adding concentration dependence will increase the accuracy of the model, but also add significant computational expense.

**Reply by MirAbbasAli2A** [reply]
@DrSOKane I do see where you are coming from. However, the part that troubles me is how can we assume the parameter set is the same when the same degree of each degradation mechanism is not observed for each input condition. For example, how can we assume that the beta value is the same for a 1C 100% DOD vs. a 0.5C 60% DOD. The former situation has way more stress-driven LAM vs. the latter? Is this difference just supposed to be accounted for by the hydrostatic stress? This also applies to any of the other fitted parameters (e.g how can the plating rate be the same for a charging rate that is 0.5C vs 2C?)

I tried doing the exercise of extending the model to have concentration dependent solid phase diffusivities over [here](https://github.com/pybamm-team/PyBaMM/discussions/3537) with the ORegan2022 definition. However, like you said, it takes about 20 min to complete 1 cycle despite me meshing it just like Oregan2022 did and even then, it runs for 20 cycles before failing.
